src/annotation/annotations_app/views/annotations.py

```diff
@@ -37,41 +36,45 @@ def annotations_list():
             application/json:
               schema: AnnotationListSchema
     """
-    with db_session() as db:
-        # TODO: add 404s for collection or image mismatch
-        args = request.args
-        image_id = args.get("image_id")
-        collection_id = args.get("collection_id")
-
-        logging.info(
-            f"GET annotations collection_id: {collection_id} image_id: {image_id}"
-        )
-        queryset = db.query(Annotation).filter(
-            and_(
-                Image.id == image_id if image_id is not None else True,
-                Annotation.image_id == Image.id,
-                Image.collections.any(
-                    and_(
-                        ImageCollection.id == collection_id
-                        if collection_id is not None
-                        else True,
-                        ImageCollection.user_id == current_user.id,
-                    )
-                ),
-            )
+    # TODO: add 404s for collection or image mismatch
+    args = request.args
+    image_id = args.get("image_id")
+    collection_id = args.get("collection_id")
+
+    if image_id:
+        # by retrieving the image ID we ensure it's available
+        Image.get_image_or_abort(image_id, current_user)
```

Comment:
Just a comment that the image id could be optional, in which case the query should return all the annotations in the collection. This could be useful for exporting training data.

Reply:
should be working I guess

Reply:
checked it, works fine both with access limitations (to image and collection) and filters; plus it's possible to display all your collections at once without any filters (apart of 'my own only' filter which can't be removed)